package NKikimrColumnShardColumnDefaults;

message TScalar {
    message TTimestamp {
        optional uint64 Value = 1;
        optional uint32 Unit = 2;
    }
    oneof Value {
        bool Bool = 1;
        uint32 Uint8 = 2;
        uint32 Uint16 = 3;
        uint32 Uint32 = 4;
        uint64 Uint64 = 5;

        int32 Int8 = 6;
        int32 Int16 = 7;
        int32 Int32 = 8;
        int64 Int64 = 9;

        double Double = 10;

        TTimestamp Timestamp = 11;

        float Float = 12;
        string String = 13;
    }
}

message TColumnDefault {
    optional TScalar Scalar = 1;
}
